src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -188,6 +195,101 @@ public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerial
             return default;
         }
 
+        private static bool IsNumericString(ref Utf8JsonReader reader)
+        {
+            ReadOnlySpan<byte> span = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+
+            // Numeric value with Leading and trailing whitespaces should be treated as valid based on behavior of Enum.TryParse()
+#if NETCOREAPP
+            span = span.Trim((byte)' ');
+#else
+            int leftIdx = 0;
+            for (; leftIdx < span.Length; leftIdx++)
+            {
+                if (span[leftIdx] != ' ')
+                {
+                    break;
+                }
+            }
+
+            int rightIdx = span.Length - 1;
+            for (; rightIdx > leftIdx; rightIdx--)
+            {
+                if (span[rightIdx] != ' ')
+                {
+                    break;
+                }
+            }
+
+            span = span.Slice(leftIdx, rightIdx - leftIdx + 1);
+#endif
+            if (span.Length > 1 && span[0] == '+')
+            {
+                if ((uint)(span[1] - '0') > 9)
+                {
+                    return false;
+                }
+
+                // Utf8Parser.TryParse() doesn't treat "+1" as a valid unsigned numeric value but Enum.TryParse() does, so need to remove '+' to align behavior
+                span = span.Slice(1);
+            }
+
+            switch (s_enumTypeCode)
+            {
+                // Switch cases ordered by expected frequency
```

Comment:
I don't think this switch statement is necessary. You just need to check that the string contains a sign plus numbers. If the numeric string is too large then it will be rejected by the `Enum.Parse` method downstream.

Reply:
Use regex now, also based on [comment](https://github.com/dotnet/runtime/pull/79432#discussion_r1339031166)